### PR TITLE
EVMOS spec from Stake Village

### PR DESCRIPTION
**Pull request by semalis**
Hello, Lava Foundation!
Let me introduce myself. My name is Sergey - the leader of the "Stake Village" team.
I have been with the Lava for over 2 years. During this time, I have been actively contributing to the project and participating in its development, providing various services:
- [Discord BOT](https://discord.com/channels/963778337904427018/1204206450818293771) - Monitoring provider status, which helps to track the status of providers and generally increases the reliability of the Lava project.
- [Services](https://stakevillage.net/en/) (Guides, API, RPC, snapshots)
- Support for Providers in both Testnet and Mainnet
Own [Telegram](https://t.me/stakevillage) community of providers, where I provide support for providers, technical details and solve problems of launching providers.

I was one of the first to launch the Evmos RPC archive node. And for a long time I was the only provider who supported it.
Participated and was awarded in "Evmos endpoints benchmarking".
Conducted tests and provided useful data for the team, necessary for analyzing resources for various requests.
Was an active validator of the Evmos network until the number of validators was reduced to 20.

I already have experience in creating new specifications, I made specifications for the Holesky (Ethereum) test network.

Discord: semalist
Commission: 3.5%

- +1% Supporting and running a provider on testnet.
-  +0.5% Inherit Cosmos spec.
- +1% Benchmarking Compute Units (CU) and timeouts.
- +1%  Active support to Providers, if more than 20 providers are retained over 3 months

### CODE REVIEW

File: mainnet-1/specs/evmos.json [hunk 1]
```diff
@@ -0,0 +1,2213 @@
...
+        "average_block_time": 6500,
+        "allowed_block_lag_for_qos_sync": "2",
+        "block_last_updated": "0",
+        "min_stake_provider": {
+          "denom": "ulava",
+          "amount": "15000000000"
+        },
+        "providers_types": "dynamic",
+        "imports": [
+          "COSMOSSDK",
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:23]: evmos is running cosmos v50
also needs to inherit from cosmossdk50

File: mainnet-1/specs/evmos.json [hunk 2]
```diff
@@ -0,0 +1,2213 @@
...
+    "specs": [
+      {
+        "index": "EVMOS",
+        "name": "evmos mainnet",
+        "enabled": true,
+        "reliability_threshold": 268435455,
+        "data_reliability_enabled": true,
+        "block_distance_for_finalized_data": 0,
+        "blocks_in_finalization_proof": 1,
+        "average_block_time": 6500,
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:14]: average blocktime for evmos is 3.9s

File: mainnet-1/specs/evmos.json [hunk 3]
```diff
@@ -0,0 +1,2213 @@
...
+                ]
+              },
+              {
+                "name": "pruning",
+                "parse_directive": null,
+                "values": [
+                  {
+                    "extension": "",
+                    "expected_value": "",
+                    "latest_distance": 14900,
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:1117]: this needs to be the number of blocks in 24h
so: 24*60*60/3.9

File: mainnet-1/specs/evmos.json [hunk 4]
```diff
@@ -0,0 +1,2213 @@
...
+              }
+            ],
+            "headers": [],
+            "inheritance_apis": [],
+            "parse_directives": [],
+            "extensions": [
+              {
+                "name": "archive",
+                "rule": {
+                  "block": 9990
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:1080]: this needs to be the number of blocks in 23H

File: mainnet-1/specs/evmos.json [hunk 5]
```diff
@@ -0,0 +1,2213 @@
...
+                ]
+              },
+              {
+                "name": "pruning",
+                "parse_directive": null,
+                "values": [
+                  {
+                    "extension": "",
+                    "expected_value": "",
+                    "latest_distance": 14900,
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:1940]: same here

File: mainnet-1/specs/evmos.json [hunk 6]
```diff
@@ -0,0 +1,2213 @@
...
+            },
+            "apis": [],
+            "headers": [],
+            "inheritance_apis": [],
+            "parse_directives": [],
+            "extensions": [
+              {
+                "name": "archive",
+                "rule": {
+                  "block": 9990
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:1903]: same here

File: mainnet-1/specs/evmos.json [hunk 7]
```diff
@@ -0,0 +1,2213 @@
...
+                    "default_value": "",
+                    "encoding": ""
+                  },
+                  "api_name": ""
+              },
+                "values": [
+                  {
+                    "extension": "",
+                    "expected_value": "",
+                    "latest_distance": 14900,
```
Comment (Yaroms) [inline mainnet-1/specs/evmos.json:2004]: same here

**Comment by Yaroms** [comment]
all good now i approve

**Comment by Yaroms** [comment]
please make sure you have the following up to date list of supported apis:
rest:
```

"/evmos/erc20/v1/token_pairs"
"/evmos/evm/v1/params"
"/evmos/inflation/v1/params"
"/evmos/evm/v1/estimate_gas"
"/evmos/evm/v1/trace_block"
"/evmos/evm/v1/balances/{address}"
"/evmos/evm/v1/account/{address}"
"/evmos/vesting/v2/balances/{address}"
"/evmos/erc20/v1/token_pairs/{token}"
"/evmos/evm/v1/storage/{address}/{key}"
"/evmos/inflation/v1/period"
"/evmos/evm/v1/trace_tx"
"/evmos/inflation/v1/epoch_mint_provision"
"/evmos/inflation/v1/inflation_rate"
"/evmos/epochs/v1/current_epoch"
"/evmos/feemarket/v1/block_gas"
"/evmos/erc20/v1/params"
"/evmos/feemarket/v1/base_fee"
"/evmos/evm/v1/validator_account/{cons_address}"
"/evmos/feemarket/v1/params"
"/evmos/epochs/v1/epochs"
"/evmos/evm/v1/cosmos_account/{address}"
"/evmos/inflation/v1/circulating_supply"
"/evmos/evm/v1/eth_call"
"/evmos/evm/v1/codes/{address}"
"/evmos/inflation/v1/skipped_epochs"
"/evmos/evm/v1/base_fee"
```
grpc:
```

"evmos.epochs.v1.Query/CurrentEpoch"
"ethermint.evm.v1.Query/Code"
"evmos.inflation.v1.Query/InflationRate"
"ethermint.evm.v1.Query/Params"
"evmos.epochs.v1.Query/EpochInfos"
"ethermint.evm.v1.Query/CosmosAccount"
"ethermint.feemarket.v1.Query/BlockGas"
"evmos.vesting.v2.Query/Balances"
"evmos.erc20.v1.Query/Params"
"evmos.inflation.v1.Query/SkippedEpochs"
"evmos.erc20.v1.Query/TokenPair"
"ethermint.feemarket.v1.Query/BaseFee"
"ethermint.evm.v1.Query/ValidatorAccount"
"ethermint.evm.v1.Query/TraceBlock"
"evmos.inflation.v1.Query/Params"
"ethermint.evm.v1.Query/Account"
"ethermint.evm.v1.Query/TraceTx"
"ethermint.evm.v1.Query/EthCall"
"evmos.inflation.v1.Query/EpochMintProvision"
"ethermint.evm.v1.Query/EstimateGas"
"evmos.erc20.v1.Query/TokenPairs"
"evmos.inflation.v1.Query/CirculatingSupply"
"evmos.inflation.v1.Query/Period"
"ethermint.feemarket.v1.Query/Params"
"ethermint.evm.v1.Query/BaseFee"
"ethermint.evm.v1.Query/Storage"
"ethermint.evm.v1.Query/Balance"
```

**Comment by semalis** [comment]
Added an Ethermint specification that supports additional apis:

**Pull request closed without merging**